lib/screens/creator_hub/creator_hub_view_model.dart

```diff
@@ -33,24 +40,90 @@ class CreatorHubViewModel extends ChangeNotifier {
     notifyListeners();
   }
 
+  final List<NFT> _publishedNFTsList = [];
+
+  List<NFT> get publishedNFTsList => _publishedNFTsList;
+
+  String? getCookbookIdFromLocalDatasource() {
+    return repository.getCookbookId();
+  }
+
+  getTotalForSale() {
+    forSaleCount = 0;
+    for (int i = 0; i < _publishedNFTsList.length; i++) {
+      if (publishedNFTsList[i].isEnabled && publishedNFTsList[i].amountMinted < publishedNFTsList[i].quantity) {
+        forSaleCount++;
+      }
+    }
+  }
+
+  Future<void> getPublishAndDraftData() async {
+
+    await getRecipesList();
+    await getDraftsList();
+
+    getTotalForSale();
+    notifyListeners();
+  }
+
+  Future<void> getRecipesList() async {
+
+    final cookBookId = getCookbookIdFromLocalDatasource();
+    if (cookBookId == null) {
+      return;
+    }
+
+    final recipesListEither = await repository.getRecipesBasedOnCookBookId(cookBookId: cookBookId);
+
+    if (recipesListEither.isLeft()) {
+      return;
+    }
+
+    final recipesList = recipesListEither.getOrElse(() => []);
+    _publishedNFTsList.clear();
+    if (recipesList.isNotEmpty) {
+      for (final recipe in recipesList) {
+        final nft = NFT.fromRecipe(recipe);
+
+        _publishedNFTsList.add(nft);
+      }
+    }
+
+    publishedRecipeLength = publishedNFTsList.length;
+  }
+
+  List<NFT> nftList = [];
+
   Future<void> getDraftsList() async {
     final loading = Loading().showLoading(message: "loading ...");
 
-    nftList = await localDataSource.getNfts();
+    final getNftResponse = await repository.getNfts();
+
+    if (getNftResponse.isLeft()) {
+      loading.dismiss();
+
+      navigatorKey.currentState!.overlay!.context.show(message: "something_wrong".tr());
+
+      return;
+    }
+
+    nftList = getNftResponse.getOrElse(() => []);
 
     loading.dismiss();
 
     notifyListeners();
   }
 
+  Future<void> deleteNft(int? id) async {
+    final deleteNftResponse = await repository.deleteNft(id!);
 
-  Future<void> deleteDraft(int? id) async {
-    bool success = await localDataSource.deleteNft(id!);
+    if (deleteNftResponse.isLeft()) {
+      navigatorKey.currentState!.overlay!.context.show(message: "delete_error".tr());
+    }
+    else{
```

Comment:
Kindly remove this else here
Add return statement on line 123 
Use extension message for showing snackbar.